registry/coder/modules/amazon-q/README.md

```diff
@@ -1,121 +1,425 @@
+  EOT
+
+  post_install_script = <<-EOT
+    #!/bin/bash
+    echo "Configuring Amazon Q settings..."
+    # Custom configuration commands
+    q settings chat.model claude-3-sonnet
+  EOT
 }
 ```
 
-### Enable task reporting (experimental)
+### Specific Version Installation
 
 ```tf
 module "amazon-q" {
-  source                  = "registry.coder.com/coder/amazon-q/coder"
-  version                 = "1.1.2"
-  agent_id                = coder_agent.example.id
-  experiment_auth_tarball = var.amazon_q_auth_tarball
-  experiment_report_tasks = true
+  source           = "registry.coder.com/coder/amazon-q/coder"
+  version          = "2.0.0"
+  agent_id         = coder_agent.example.id
+  auth_tarball     = var.amazon_q_auth_tarball
+  amazon_q_version = "1.14.0" # Specific version
+  install_amazon_q = true
 }
 ```
 
-### Run custom scripts before/after install
+### Custom Agent Configuration
 
 ```tf
 module "amazon-q" {
-  source                         = "registry.coder.com/coder/amazon-q/coder"
-  version                        = "1.1.2"
-  agent_id                       = coder_agent.example.id
-  experiment_auth_tarball        = var.amazon_q_auth_tarball
-  experiment_pre_install_script  = "echo Pre-install!"
-  experiment_post_install_script = "echo Post-install!"
+  source       = "registry.coder.com/coder/amazon-q/coder"
+  version      = "2.0.0"
+  agent_id     = coder_agent.example.id
+  auth_tarball = var.amazon_q_auth_tarball
+
+  agent_config = <<-EOT
+    {
+      "name": "custom-agent",
+      "description": "Custom Amazon Q agent for my workspace",
+      "prompt": "You are a specialized DevOps assistant...",
+      "tools": ["fs_read", "fs_write", "execute_bash", "use_aws"]
+    }
+  EOT
 }
 ```
 
-## Notes
+### With Task Reporting and Custom Working Directory
+
+```tf
+module "amazon-q" {
+  source       = "registry.coder.com/coder/amazon-q/coder"
+  version      = "2.0.0"
+  agent_id     = coder_agent.example.id
+  auth_tarball = var.amazon_q_auth_tarball
+
+  # Task reporting configuration
+  report_tasks = true
+  workdir      = "/workspace/project"
+
+  # Enable CLI app alongside web app
+  cli_app              = true
+  web_app_display_name = "Amazon Q Assistant"
+  cli_app_display_name = "Q CLI"
+}
+```
+
+### With Custom AgentAPI Configuration
+
+```tf
+module "amazon-q" {
+  source       = "registry.coder.com/coder/amazon-q/coder"
+  version      = "2.0.0"
+  agent_id     = coder_agent.example.id
+  auth_tarball = var.amazon_q_auth_tarball
+
+  # AgentAPI configuration for environments without wildcard access
+  agentapi_chat_based_path = true
+  agentapi_version         = "v0.6.1"
+}
+```
+
+### UI Customization
+
+```tf
+module "amazon-q" {
+  source       = "registry.coder.com/coder/amazon-q/coder"
+  version      = "2.0.0"
+  agent_id     = coder_agent.example.id
+  auth_tarball = var.amazon_q_auth_tarball
+
+  # UI configuration
+  order = 1
+  group = "AI Tools"
+  icon  = "/icon/custom-amazon-q.svg"
+}
+```
```

Comment:
All examples above, and this can add a sentence or two to describe what they do.
For example here it will be helpful to know what UI configuration is? My guess is its Amazon-q web app? OR is it the chat app exposed by agentapi?